crabs/detection_tracking/detection_utils.py

```diff
@@ -226,65 +229,243 @@ def calculate_iou(box1, box2) -> float:
     return iou
 
 
-def draw_gt_boxes(
-    ground_truths: dict,
-    frame_number: int,
+def count_identity_switches(
+    prev_frame: Optional[List[List[float]]], current_frame: List[List[float]]
+) -> int:
+    """
+    Count the number of identity switches between two sets of object IDs.
+    """
+    if prev_frame is None:
+        # If there are no previous frame IDs, return 0 switches
+        return 0
+
+    # Convert tracked boxes to tuples for comparison
+    prev_frame_tuples = [tuple(box) for box in prev_frame]
+    current_frame_tuples = [tuple(box) for box in current_frame]
+
+    # Create dictionaries to track object IDs in each frame
+    id_to_index_prev = {id_: i for i, id_ in enumerate(prev_frame_tuples)}
+    id_to_index_current = {
+        id_: i for i, id_ in enumerate(current_frame_tuples)
+    }
+
+    # Initialize count of identity switches
+    num_switches = 0
+
+    # Loop through object IDs in the current frame
+    for id_current, index_current in id_to_index_current.items():
+        # Check if the object ID exists in the previous frame
+        if id_current in id_to_index_prev:
+            # Get the corresponding index in the previous frame
+            index_prev = id_to_index_prev[id_current]
+            # If the index is different, it indicates an identity switch
+            if index_current != index_prev:
+                num_switches += 1
+
+    return num_switches
+
+
+def evaluate_mota(
+    gt_boxes: np.ndarray,
     tracked_boxes: np.ndarray,
     iou_threshold: float,
-    frame_copy: np.ndarray,
+    prev_frame: Union[Optional[List[List[float]]], None],
```

Comment:
doesn't `Optional` include `None`?
```suggestion
    prev_frame: Optional[List[List[float]]],
```

Reply:
yes, my bad overlook this. Thank you